### ISSUE TITLE
deprecation rule should consider which overload is being used for warning

### ISSUE THREAD

**Issue by Kovensky** [palantir/tslint#3159]
### Bug Report

- __TSLint version__: 5.6.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import $ from 'jquery'
$().toggleClass('className')
```

with `tslint.json` configuration:

```json
{ "rules": { "deprecation": true } }
```

#### Actual behavior

`deprecation: toggleClass is deprecated: 3.0`

#### Expected behavior

Only the `toggleClass(state?: boolean): this` overload should cause the deprecation warning, as that is the only overload that is `@deprecated`.

**Comment by ajafff** [comment]
That's already fixed on master by #2883. I try to get a release out this week.
